test/contracts/workers/set_worker.js

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 OST.com Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Test: set_workers.js
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const workers_utils = require('./workers_utils.js');
+const Workers       = artifacts.require('./Workers.sol');
+
+///
+/// Test stories
+///
+/// fails to set worker if worker address is 0
+/// fails to set worker if _deactivationHeight is equal to current block number
+/// fails to set worker if _deactivationHeight is less than current block number
+/// fails to set worker if sender is not opsAddress
+/// pass to set worker if worker is not already set
+/// pass to set worker if worker was already set
+
+module.exports.perform = (accounts) => {
+  const opsAddress          = accounts[1],
+        worker1Address       =  accounts[2],
+        worker2Address       =  accounts[3],
+        height1  = new workers_utils.bigNumber(500),
+        height2  = new workers_utils.bigNumber(1000);
+        
+  before(async () => {
+
+    workers = await Workers.new();
+    assert.ok(await workers.setOpsAddress(opsAddress));    
+
+  });
+
+
+  it('fails to set worker if worker address is 0', async () => {
+
+    await workers_utils.utils.expectThrow(workers.setWorker.call(0, height1, { from: opsAddress }));    
+
+  });
+
+
+  it('fails to set worker if _deactivationHeight is equal to current block number', async () => {    
+
+    await workers_utils.utils.expectThrow(workers.setWorker.call(0, web3.eth.blockNumber, { from: opsAddress }));    
```

Comment:
worker is `0`?

Reply:
This test is very tricky to test. In fact this test should pass instead of fail.
When I call 
```
workers.setWorker.call(worker1Address, web3.eth.blockNumber, { from: opsAddress })
```
This works fine.

But then I call 
```
response = await workers.setWorker(worker1Address, web3.eth.blockNumber, { from: opsAddress });
 ```
Its not triggering ```  WorkerSet ``` event

Looks like at the below line
```
uint256 remainingHeight = _deactivationHeight - block.number;
```
remainingHeight is going negative for uint256 

Please suggest how to test